__tests__/app.test.js

```diff
@@ -101,4 +100,71 @@ describe("/api/articles/:article_id/comments", () => {
         expect(body.msg).toBe("Bad request");
       });
   });
+//   test("GET: 200 responds with an empty array for an article id that exists but has no comments", () => {
+//     return request(app)
+//       .get("/api/articles/7/comments")
+//       .expect(200)
+//       .then(({ body }) => {
+//         console.log(body, "<--- body in test");
+//         expect(body.comment).toEqual([]);
+//       });
+//   });
+});
+
+describe("/api/articles/:article_id/comments", () => {
+  test("POST: 201 responds with the new posted comment", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({
+        author: "butter_bridge",
+        body: "What is this article about?",
+      })
+      .expect(201)
+      .then(({ body }) => {
+        const created_at = body.comment[0].created_at;
+        expect(body.comment[0]).toMatchObject({
+          comment_id: 19,
+          body: "What is this article about?",
+          article_id: 1,
+          author: "butter_bridge",
+          votes: 0,
+          created_at: created_at,
+        });
+      });
+  });
+  test("POST: 404 sends an appropriate error status alongside an error message when given a valid but non-existent id", () => {
+    return request(app)
+      .post("/api/articles/1000/comments")
+      .send({
+        author: "butter_bridge",
+        body: "What is this article about?",
+      })
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Article does not exist");
+      });
+  });
+  test("POST: 400 sends an appropriate error status alongside an error message when given an invalid id", () => {
+    return request(app)
+      .post("/api/articles/invalid-id/comments")
+      .send({
+        author: "butter_bridge",
+        body: "What is this article about?",
+      })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+  test("POST: 400 sends an appropriate error status alongside an error message when given an invalid id", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({
+        author: "butter_bridge",
+      })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
```

Comment:
Please add another test (400) where the request is missing required field(s), e.g. no username or body properties.